src/Stratis.Bitcoin.Features.PoA.Tests/PoAHeaderDifficultyRuleTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NBitcoin;
+using NBitcoin.BouncyCastle.Math;
+using Stratis.Bitcoin.Features.PoA.ConsensusRules;
+using Stratis.Bitcoin.Tests.Common;
+using Xunit;
+
+namespace Stratis.Bitcoin.Features.PoA.Tests
+{
+    public class PoAHeaderDifficultyRuleTests
+    {
+        [Fact]
+        public void CumulativeWorkForALotOfBlocksIsLowerThanMaxValue()
+        {
+            Target defaultTarget = PoAHeaderDifficultyRule.PoABlockDifficulty;
+            int times = 110_000_000;
+
+            long result = defaultTarget.ToCompact() * times;
+
+            Assert.True(result < long.MaxValue);
+        }
+
+        [Fact]
+        public void ChainworkIsIncreasedBySameConstantValue()
+        {
+            BigInteger pow256 = BigInteger.ValueOf(2).Pow(256);
+            Target defaultTarget = PoAHeaderDifficultyRule.PoABlockDifficulty;
+            int headersCount = 200;
+
+            List<ChainedHeader> headers = ChainedHeadersHelper.CreateConsecutiveHeaders(headersCount, null, false, defaultTarget);
+
+            foreach (ChainedHeader chainedHeader in headers)
+                chainedHeader.InvokeMethod("CalculateChainWork");
```

Comment:
Not to do with your change, but I think this method `CalculateChainWork` should be made public and should be returning a `BigInteger` instead of assigning the `this.chainWork` field (so you 'd have `this.chainwork = CalculateChainWork()`).
Effectively, this method lies about what it's doing. It says it's calculating, but there is an undocumented side-effect.

Reply:
Agreed with your point. 
That call was when I was investigating why the test doesn't work. Found out it wasn't an issue but forgot to remove this call